lib/core-common/src/types.ts

```diff
@@ -234,22 +234,22 @@ export interface StorybookConfigOptions {
 
 export type Options = LoadOptions & StorybookConfigOptions & CLIOptions & BuilderOptions;
 
-export interface Builder<Config, Stats> {
+export interface Builder<Config, BuilderStats extends Stats | null> {
```

Comment:
can you explain the `| null` part? everything else makes sense to me

Reply:
The `null` part here is so that custom builders can disable use of `stats`. Here is how `builder-vite` is utilizing this interface: https://github.com/storybookjs/builder-vite/blob/6338427ebaf4152ae098f5a06dc065bf774d9186/packages/builder-vite/index.ts#L14-L17.

As this is a required parameter of interface generics, `null` is used to disable it. Should we make it optional, and default to `Stats`? Also make `start` method's output `{ stats }` optional:

```diff
-export interface Builder<Config, BuilderStats extends Stats | null> {
+export interface Builder<Config, BuilderStats extends Stats = Stats> {
    getConfig: (options: Options) => Promise<Config>;
    start: (args: {
        options: Options;
        startTime: ReturnType<typeof process.hrtime>;
        router: Router;
        server: Server;
    }) => Promise<void | {
-       stats: BuilderStats;
+       stats?: BuilderStats;
        totalTime: ReturnType<typeof process.hrtime>;
        bail: (e?: Error) => Promise<void>;
    }>;
```

Now custom builders could simply ignore the stats when using `Builder` interface:

```ts
export type ViteBuilder = Builder<UserConfig>;
```